src/main/java/teammates/common/util/LogHelper.java

```diff
@@ -0,0 +1,128 @@
+        query.batchSize(1000);
+        query.minLogLevel(LogLevel.INFO);
+        if (startTime != null) {
+            query.startTimeMillis(startTime);
+        }
+        if (endTime != null) {
+            query.endTimeMillis(endTime);
+        }
+        query.majorVersionIds(getVersionIdsForQuery(versionsToQuery));
+        if (offset != null && !offset.equals("null")) {
+            query.offset(offset);
+        }
+    }
+    
+    /**
+     * Selects versions for query. If versions are not specified, it will return 
+     * default versions used for query.
+     */
+    private List<String> getVersionIdsForQuery(List<String> versions) {
+        boolean isVersionSpecifiedInRequest = (versions != null && !versions.isEmpty());
+        if (isVersionSpecifiedInRequest) {
+            return versions;
+        }
+        return getDefaultVersionIdsForQuery();
+    }
+    
+    /**
+     * Gets a list of versions, including the current version and 5 preceding versions (if available).
+     * @return a list of default versions for query.
+     */
+    private List<String> getDefaultVersionIdsForQuery() {
+        ModulesService modulesService = ModulesServiceFactory.getModulesService();
+        String[] versionList = (String[]) modulesService.getVersions(null).toArray(); // null == default module
+        String currentVersion = modulesService.getCurrentVersion();
+        int currentVersionIndex;
+        List<String> defaultVersions = new ArrayList<String>();
+        try {
+            currentVersionIndex = getCurrentVersionIndex(versionList, currentVersion);
+            defaultVersions = getNextFewVersions(versionList, currentVersionIndex);
+        } catch (InvalidParametersException e) {
+            defaultVersions.add(currentVersion);
+            e.printStackTrace();
+        }
+        return defaultVersions;
+    }
+
+    /**
+     * Finds the current version then get at most 5 versions below it.
+     * @param currentVersionIndex starting position to get versions to query
+     */
+    private List<String> getNextFewVersions(String[] versionList, int currentVersionIndex) {
+        List<String> result = new ArrayList<String>();
+        for(int i = currentVersionIndex; i < versionList.length; i++) {
+            result.add(versionList[i]);
+            if (result.size() >= MAX_PAST_VERSIONS_TO_QUERY) {
+                return result;
+            }
+        }
+        return result;
+    }
+
+    /**
+     * Finds the index of the current version in the given list.
+     * @throws InvalidParametersException when the current version is not found
+     */
+    private int getCurrentVersionIndex(String[] versionList, String currentVersion) 
+                    throws InvalidParametersException {
+        for(int i = 0; i < versionList.length; i++) {
+            if (versionList[i].equals(currentVersion)) return i;
+        }
+        throw new InvalidParametersException("The current version is not found!");
+    }
```

Comment:
Doesn't `List<String> versionList` have a built-in function to achieve this?
To convert from `Set<String>` you just need to do this:
```java
Set<String> set = ... // get the set
List<String> list = new ArrayList<String>(set);
```